st3/lsp_utils/npm_client_handler.py

```diff
@@ -59,51 +92,77 @@ def setup(cls) -> None:
         if not cls.__server:
             cls.__server = ServerNpmResource(cls.package_name, cls.server_directory, cls.server_binary_path)
         cls.__server.setup()
+        cls.settings_filename = '{}.sublime-settings'.format(cls.package_name)
 
     @classmethod
     def cleanup(cls) -> None:
         if cls.__server:
             cls.__server.cleanup()
 
-    @property
-    def name(self) -> str:
-        return self.package_name.lower()  # type: ignore
-
-    @property
-    def config(self) -> ClientConfig:
-        assert self.__server
+    @classmethod
+    def config(cls) -> ClientConfig:
+        if cls.__server is None:
+            cls.setup()
 
         configuration = {
             'enabled': True,
-            'command': ['node', self.__server.binary_path] + self.get_binary_arguments(),
+            'command': ['node', cls.__server.binary_path] + cls.get_binary_arguments(),
         }
 
-        configuration.update(self._read_configuration())
-        self.on_client_configuration_ready(configuration)
-        return read_client_config(self.name, configuration)
+        configuration.update(cls._read_configuration())
+        cls.on_client_configuration_ready(configuration)
+        return read_client_config(cls.name, configuration)
```

Comment:
That could be simplified to just return configuration maybe?
Calling code can create config itself (it knows the name).